README.md

```diff
@@ -1,12 +1,111 @@
+
+# Build on AWS
+packer build -var "install_pass=${INSTALL_PASS}" resources/windows/windows.pkr.hcl
+```
+
+This will create a new AMI, and replace the existing AMI if present.
+
+### Debugging
+
+To debug a failing build you can use the `-on-error=ask` flag:
+
+```
+packer build -var "install_pass=${INSTALL_PASS}" -on-error=ask resources/windows/windows.pkr.hcl
+```
+
+You can remote into the machine via WinRM tooling, potentially via Metasploit:
+
+```msf
+msf6 auxiliary(scanner/winrm/winrm_login) > run rhost=50.18.26.233 username=Administrator password=MJ72x)O7R3D-96VbAPX).0M%nlZv9bHP rport=5986
+
+[!] No active DB -- Credential data will not be saved!
+[+] 50.18.26.233:5986 - Login Successful: WORKSTATION\Administrator:MJ72x)O7R3D-96VbAPX).0M%nlZv9bHP
+[*] Command shell session 1 opened (x.x.x.x:52424 -> x.x.x.x:5986) at 2023-11-16 12:44:46 +0000
+[*] Scanned 1 of 1 hosts (100% complete)
+[*] Auxiliary module execution completed
+
+msf6 auxiliary(scanner/winrm/winrm_login) > sessions -i -1
+[*] Starting interaction with 1...
+
+Microsoft Windows [Version 10.0.17763.4974]
+(c) 2018 Microsoft Corporation. All rights reserved.
+
+C:\Users\Administrator>
+```
+
+The `-debug` flag can be used to pause at each step as well, which will extract and print the remote EC2 build password directly
+not just the temporary Administrator account that is created:
+
+```
+$ packer build -var "install_pass=${INSTALL_PASS}" -debug -on-error=ask resources/windows/windows.pkr.hcl
+...
+==> win-builder-base.amazon-ebs.metasploit-windows-builder: Waiting for instance (i-0f18b8be1c11b0893) to become ready...
+win-builder-base.amazon-ebs.metasploit-windows-builder: Public DNS: ec2-50-18-26-233.us-west-1.compute.amazonaws.com
+win-builder-base.amazon-ebs.metasploit-windows-builder: Public IP: 50.18.26.233
+win-builder-base.amazon-ebs.metasploit-windows-builder: Private IP: 172.31.13.97
+==> win-builder-base.amazon-ebs.metasploit-windows-builder: Pausing after run of step 'StepRunSourceInstance'. Press enter to continue.
+==> win-builder-base.amazon-ebs.metasploit-windows-builder: Waiting for auto-generated password for instance...
+win-builder-base.amazon-ebs.metasploit-windows-builder: It is normal for this process to take up to 15 minutes,
+win-builder-base.amazon-ebs.metasploit-windows-builder: but it usually takes around 5. Please wait.
+win-builder-base.amazon-ebs.metasploit-windows-builder:
+win-builder-base.amazon-ebs.metasploit-windows-builder: Password retrieved!
+win-builder-base.amazon-ebs.metasploit-windows-builder: Password (since debug is enabled): MJ72x)O7R3D-96VbAPX).0M%nlZv9bHP
+==> win-builder-base.amazon-ebs.metasploit-windows-builder: Pausing after run of step 'StepGetPassword'. Press enter to continue.
+```
+
+See more details in the [documentation](https://github.com/hashicorp/packer/blob/c245b1fb7c87fdf2e655887d49f8ad75c59b7e2b/website/content/docs/debugging.mdx#L9)
+
+### Security considerations
```

Comment:
Thank you for adding the warnings here, these are all salient points that needed good docs.